### DISCUSSION TITLE
I am very disappointed - or annoyed

### DISCUSSION THREAD

**Discussion by geraldFS** [omersiar/esp-rfid#622] [General]
Now I'm getting fed up, sorry!
But with every new attempt it gets worse, I thought now a breath of fresh air would come in but exactly the opposite happened.

I've been using this solution as a front door for a long time, and I was happy when new momentum came in recently, now it's even more possible.

2.0.RC installed

- RFID will no longer be read. -> Reader Wiegang Reader - new ordered no change.
- Board restarts regularly
- MQTT cruel
- Log files are no longer written

Back to old version - the RFID keys are no longer accepted.

I have now installed an alternative.

Somehow it seems to me that it was optimized badly!

**Comment by matjack1** [comment]
Hey @geraldFS let me start by saying that I feel your pain. I've spent a lot of time on this project and I agree that it's not working as well as we all want.

But I cannot help you with this kind of bug reports.

If you still want to use this project and you want to help, we need to go a bit in the details of your problems.

1. Wiegand reader not working: I'm using Wiegand readers and it works, it might be due to a configuration problem/usage of a pin reserved to other purposes? If you want to share the configuration file that you are using I can have a look.
2. How frequently the board restarts? Which board are you using? What kind of usage are you doing? I have three boards used only via MQTT and they are mostly stable, they might restart once every few days. The web UI sometimes crashes, but if you wait it should fetch all data eventually. We can dig deeper here if you want.
3. What do you mean by MQTT cruel? Do you refer to the API change? If so, yes the v2 is a breaking change, but to support Home Assistant we had introduced a lot of stuff, also the API was grown organically and needed a bit of reordering. If this is a deal breaker, then I'm sorry, but it wasn't really comprehensible for newcomers :(
4. Which logfiles are not written? I have recently checked and it's working as expected to me.

In general, I've tried my best to reduce the instability, but with the growing codebase the web UI is more and more unstable due to the websocket part being unreliable. Unfortunately there's not much we can do there without massive rewriting. I think it's not realistic to add a lot more code in here without having more frequent problems there.

**Comment by matjack1** [comment]
Hey @geraldFS not sure if you are still interested, but if you were using esp-rfid in access point mode (not connected to a WiFi network), then you might have hit a bug that is now fixed. Just to let you know that it might have been just that :)
